Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2381 @@
+/*           not-activated - query profiles for not activated images   */
+/*           Any other values, or no parm specified will default to    */
+/*           operating, value is case sensitive                        */
+/*        -I indicate running out of ISV REXX environment,             */
+/*           default is TSO/E                                          */
+/*        -V turn on additional verbose JSON tracing                   */
+/*                                                                     */
+/*        EX 'HWI.HWIREST.REXX(RXCRYPT1))'                             */
+/*            '-D HWI.CRYPTO -I'                                       */
+/*            data set is HWI.CRYPTO, run out of ISV REXX,             */
+/*            if successful will create HWI.CRYPTO(LOCAL)              */
+/*                                                                     */
+/* DEPENDENCIES:                                                       */
+/*     z16 - The crypto properties are supporeted on z16 processors    */
+/*           and above.                                                */
+/*                                                                     */
+/* NOTES:                                                              */
+/*     No recovery logic has been supplied in this sample.             */
+/*                                                                     */
+/* REFERENCE:                                                          */
+/*     See the z/OS MVS Programming: Callable Services for             */
+/*     High-Level Languages publication for more information           */
+/*     regarding the usage of HWIREST and JSON Parser APIs.            */
+/*                                                                     */
+/*                                                                     */
+/* END OF SPECIFICATIONS  * * * * * * * * * * * * * * * * * * * * * * **/
+
+MACLIB_DATASET = 'SYS1.MACLIB'
+
+TEMP_DATASET = '' /* required input via -D */
+MEMBER_NAME = 'LOCAL' /* default CPC */
+LPAR_Status = 'operating' /* default Image status */
+
+TRUE = 1
+FALSE = 0
+
+hwiHostRC = 0     /* HWIHOST rc */
+HWIHOST_ON = FALSE
+PARSER_INIT = FALSE
+ISVREXX = FALSE  /* default to TSO/E, enable via -I */
+VERBOSE = FALSE  /* JSON parser specific, enabled via -V */
+localCPC = TRUE  /* default to lpars on the LOCAL CPC, change via -C */
+
+/* Properties constants */
+cryptoauth = 'crypto-activity-cpu-counter-authorization-control'
+assignedDomainsProp = 'assigned-crypto-domains'
+assignedCryptosProp = 'assigned-cryptos'
+validst1            = 'operating'
+validst2            = 'not-operating'
+validst3            = 'not-activated'
+
+/*********************************/
+/* Get program args              */
+/*********************************/
+parse arg argString
+if GetArgs(argString) <> 0 then
+  exit -1
+
+Select
+ When LPAR_Status = validst1 then NOP
```

Comment:
this is a bit beyond me, but why does it matter what status the lpar is in? what impact does it have? if it's not in operating ....does the code need to retrieve the crypto attributes from an image activation profile or something...... trying to understand why LPAR status matters

Reply:
it was a "feature" requested by the customer, he wanted the ability to see crypto properties for not-activated lpars before activating